File: DPProgressRingExample-ObjC/ViewController.m
//
//  ViewController.m
//  DPProgressRingExample-ObjC
//
//  Created by 张鹏 on 2020/7/14.
//  Copyright © 2020 dancewithpeng@gmail.com. All rights reserved.
//

#import "ViewController.h"
#import <DPProgressRing/DPProgressRing.h>

@interface ViewController ()

@property (nonatomic, strong) ProgressRing *ring;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.ring = [[ProgressRing alloc] initWithFrame:CGRectMake(100, 100, 200, 200)];
    self.ring.isClockwise = NO;
    self.ring.maximum = 4;
    self.ring.minimum = 0;
    self.ring.width = 20;
    [self.view addSubview:self.ring];
}

- (IBAction)randomButtonDidTap:(id)sender {
    self.ring.progress = arc4random()%5;
}

- (IBAction)animationRandomButtonDidTap:(id)sender {
    [self.ring setProgress:arc4random()%5 animated:YES];
}


@end

File: DPProgressRingExample/ViewController.swift
//
//  ViewController.swift
//  DPProgressRingExample
//
//  Created by 张鹏 on 2020/7/13.
//  Copyright © 2020 dancewithpeng@gmail.com. All rights reserved.
//

import UIKit
import DPProgressRing

class ViewController: UIViewController {
    
    lazy var ring = ProgressRing(frame: CGRect(x: 100, y: 100, width: 200, height: 200))

    override func viewDidLoad() {
        super.viewDidLoad()
        
        view.addSubview(ring)
        ring.progress         = 0.45
        ring.isClockwise      = true
        ring.startAngleOffset = -CGFloat.pi/2.0
        
        ring.placeholderColor = UIColor.lightGray
        ring.progressColor    = UIColor.yellow
    }
    
    @IBAction func randomButtonDidTap(_ sender: Any) {
        ring.progress = Float.random(in: 0...1)
    }
    
    @IBAction func animatedRandomButtonDidTap(_ sender: Any) {
        ring.setProgress(Float.random(in: 0...1), animated: true)
    }
}

